main.go

```diff
@@ -18,29 +18,41 @@ var DefaultTag = "latest"
 var skipNs bool
 var pull bool
 var image string
+var dockerEndpoint = ""
 
 var rootCmd = &cobra.Command{
 	Use:   "crane CONTAINER [COMMAND] [ARG...]",
 	Short: "crane is a small utility for debugging docker containers",
 	Args:  cobra.MinimumNArgs(1),
-	Run: func(cmd *cobra.Command, args []string) {
-		client, err := gdc.NewClientFromEnv()
-		if err != nil {
-			log.Fatalf("Cannot connect to docker: %s", err)
-		}
-		img, err := internal.PrepareDebugImage(client, image, pull)
-		if err != nil {
-			log.Fatalf("Cannot prepare debug image: %s", err)
-		}
-		err = internal.RunDebugContainer(client, img, skipNs, args[0], args[1:])
-		if err != nil {
-			log.Fatalf("Problems with debug container: %s", err)
-		}
-	},
+	Run: rootCmdFunc,
+}
+
+func rootCmdFunc(cmd *cobra.Command, args []string) {
+	var client *gdc.Client
+	var err error
+	if dockerEndpoint != "" {
+		client, err = gdc.NewClient(dockerEndpoint)
+	} else {
+		client, err = gdc.NewClientFromEnv()
+	}
+	if err != nil {
+		log.Fatalf("Cannot connect to docker: %s", err)
+	}
+	img, err := internal.PrepareDebugImage(client, image, pull)
+	if err != nil {
+		log.Fatalf("Cannot prepare debug image: %s", err)
+	}
+	err = internal.RunDebugContainer(client, img, skipNs, args[0], args[1:])
+	if err != nil {
+		log.Fatalf("Problems with debug container: %s", err)
+	}
 }
 
 func main() {
 	rootCmd.PersistentFlags().StringVar(&image, "image", fmt.Sprintf("%s/%s:%s", DefaultRepo, DefaultImage, DefaultTag), "Image with debugging tools")
+	rootCmd.PersistentFlags().StringVar(&dockerEndpoint, "docker-endpoint", "", "Path or an address of the docker endpoint" +
```

Comment:
Well, considering that we pass it directly to the dockerclient library which uses it's own formatting (more or less) and calls it "docker endpoint" i think that it will be more appropriate to use their terms.

Please note that i gave a couple of examples later on, so there should not be any misunderstandings (hopefully).